spec/commands/exec_spec.rb

```diff
@@ -218,6 +218,57 @@
     expect(out).to include("bundler: exec needs a command to run")
   end
 
+  describe "with help flags" do
+    before(:each) do
+      install_gemfile <<-G
+        gem "rack"
+      G
+    end
+
+    it "shows executable's man page when --help is after the executable" do
+      bundle "exec echo --help"
+      expect(out).to include("Usage: echo [SHORT-OPTION]... [STRING]...")
+    end
+
+    it "shows executable's man page when --help is between exec and the executable" do
+      bundle "exec --help echo"
+      expect(out).to include("Usage: echo [SHORT-OPTION]... [STRING]...")
```

Comment:
I think I would expect this one to show the help/man page for `bundle exec` as it's like you were running `bundle exec --help (echo)`. And the standalone command (without bundle exec) would essentially be just echo. I.e. running  `--help echo` is not a valid command and does not show the help for echo.